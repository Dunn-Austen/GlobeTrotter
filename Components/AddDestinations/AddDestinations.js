import React, { useState, useEffect } from 'react';
import { StyleSheet, Text, TextInput, TouchableOpacity, View, ScrollView } from 'react-native'
const AddDestinations = ({ navigation, route }) => {
  // const { tripId } = route.params
  let [destinationLocation, setDestinationLocation] = useState('');
  let [startDate, setStartDate] = useState('');
  let [endDate, setEndDate] = useState('')

  return (
    <ScrollView
      contentContainerStyle={styles.container}
      keyboardShouldPersistTaps='handled'
    >
      <Text style={styles.label}>Destination</Text>
      <View style={styles.inputContainer}>
        <TextInput 
          style={styles.input} 
          name='destination'
          value={destinationLocation}
          onChangeText={(text) => setDestinationLocation(text)}
        />
      </View>
      <TouchableOpacity
        activeOpacity={.8}
        style={styles.button}
        onPress={() => navigation.navigate('Calendar')}
      >
        <Text style={{ color: '#0D47A1', fontSize: 20, fontWeight: 'bold' }}>Select Dates</Text>
      </TouchableOpacity>
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#1E88E5',
    height: '100%',
    alignItems: 'center',
    padding: 15
  },
  label: {
    color: '#fff',
    fontSize: 20
  },
  input: {
    justifyContent: 'center',
    height: '100%',
    width: '100%',
    fontSize: 20,
    paddingHorizontal: 20
  },
  inputContainer: {
    width: '100%',
    alignItems: 'center',
    backgroundColor: '#fff',
    marginBottom: 20,
    marginTop: 5,
    borderRadius: 5,
    height: 70
  },
  button: {
    backgroundColor: '#2CCB70',
    width: '100%',
    height: 50,
    borderRadius: 5,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 15
  }
})

export default AddDestinations;