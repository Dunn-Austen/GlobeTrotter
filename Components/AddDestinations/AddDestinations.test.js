import React from 'react';
import AddDestinations from './AddDestinations';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<AddDestinations />).toJSON();
  expect(tree).toMatchSnapshot();
});