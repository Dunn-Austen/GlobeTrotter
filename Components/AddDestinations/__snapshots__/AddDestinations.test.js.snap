// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1E88E5",
      "height": "100%",
      "padding": 15,
    }
  }
  keyboardShouldPersistTaps="handled"
>
  <View>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
        }
      }
    >
      Destination
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "height": 70,
          "marginBottom": 20,
          "marginTop": 5,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        name="destination"
        onChangeText={[Function]}
        placeholder="City, Country"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 20,
            "height": "100%",
            "justifyContent": "center",
            "paddingHorizontal": 20,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2CCB70",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 15,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0D47A1",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Select Dates
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
